content_templates/templates/common/child_content_builder.html

```diff
@@ -0,0 +1,59 @@
+{% import "common/macros.html" as macros %}
+{% import "common/instance_view_builder.html" as view_builder %}
+
+{% macro build_child_collections(graph, collection_uri, level) %}
+    <p class="debug-children">build_child_collections | level: {{ level }} | collection_uri: {{ collection_uri }}</p>
+    {% set select_query =
+    "PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+
+    SELECT ?member ?prefLabel
+    FROM <"~graph~">
+    {
+      values ?collection {<"~collection_uri~">}
+      ?member a skos:Collection .
+      ?collection skos:member ?member .
+      optional {
+        ?member skos:prefLabel ?prefLabel
+      }
+    }" %}
+    {% set content, error = from_endpoint(conf.default_endpoint).with_query(select_query).fetch_tabular() %}
+    {% call macros.render_fetch_results(content, error) %}
+        {% if not content.empty %}
+            <p class="debug">collection's not empty {{ collection_uri }}</p>
+            {% for row in content.itertuples() %}
+                <p class="debug">building {{ row.prefLabel }}</p>
+                {{ view_builder.build_instance_view(graph, row.member, row.prefLabel, level) }}
+                {{ build_child_collections(graph, row.member, level + 1) }}
+                {{ build_child_concepts(graph, row.member, level + 1) }}
+            {% endfor %}
+        {% endif %}
+    {% endcall %}
+{% endmacro %}
+
+
+{% macro build_child_concepts(graph, collection_uri, level) %}
+    <p class="debug-children">build_child_concepts | level: {{ level }} | collection_uri: {{ collection_uri }}</p>
+    {% set select_query =
+    "PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+
+    SELECT ?member ?prefLabel
+    FROM <"~graph~">
+    {
+      values ?collection {<"~collection_uri~">}
+      ?member a skos:Concept .
+      ?collection skos:member ?member .
+      optional {
+        ?member skos:prefLabel ?prefLabel
+      }
+    }" %}
+    {% set content, error = from_endpoint(conf.default_endpoint).with_query(select_query).fetch_tabular() %}
+    {% call macros.render_fetch_results(content, error) %}
+        {% if not content.empty %}
+            <p class="debug">concept's not empty {{ collection_uri }}</p>
```

Comment:
perhaps we can get rid of debug unless we truly need it;

Reply:
I would rather leave them for the first few deliveries since it will allow to more easily pinpoint irregularities in the report and make adjustments to it